controllers/task.go

```diff
@@ -0,0 +1,76 @@
+package controllers
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/coduno/engine/model"
+	"github.com/coduno/engine/passenger"
+	"github.com/coduno/engine/util"
+	"github.com/gorilla/mux"
+	"golang.org/x/net/context"
+	"google.golang.org/appengine/datastore"
+)
+
+// GetTaskByKey loads a task by key
+func GetTaskByKey(ctx context.Context, w http.ResponseWriter, r *http.Request) {
+	p, ok := passenger.FromContext(ctx)
+	if !ok {
+		http.Error(w, "Unauthorized", http.StatusUnauthorized)
+		return
+	}
+	if !util.CheckMethod(w, r, "GET") {
+		return
+	}
+	taskKey, err := datastore.DecodeKey(mux.Vars(r)["id"])
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	// User is a coder
+	if p.UserKey.Parent() == nil {
+		rk, err := datastore.DecodeKey(r.URL.Query()["result"][0])
+		if err != nil {
+			http.Error(w, "Key decoding err "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+		var result model.Result
+		err = datastore.Get(ctx, rk, &result)
+		if err != nil {
+			http.Error(w, "Datastore err "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+		var challenge model.Challenge
+		err = datastore.Get(ctx, result.Challenge, &challenge)
+		if err != nil {
+			http.Error(w, "Datastore err "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		emptyTime := time.Unix(0, 0)
+		updateResult := false
+		for i, val := range challenge.Tasks {
+			if taskKey.Equal(val) && result.StartTimes[i] == emptyTime {
+				result.StartTimes[i] = time.Now()
+				updateResult = true
+			}
+		}
+		if updateResult {
+			_, err = result.Save(ctx, rk)
```

Comment:
You update the result without checking that the currently logged in user is parent of the result.